src/shared/utils/filteredData.js

```diff
@@ -0,0 +1,44 @@
+const getFilterdData = (sortedData,state) => {
+  if (!state.inStock) {
+    sortedData = sortedData.filter((elm) => {
+      return elm.in_stock === true;
+    });
+   }
+  if (state.fastDelivery) {
+    sortedData = sortedData.filter((elm) => {
+      return elm.fast_delivery === true;
+    });
+  }
+  if (state.range) {
+    sortedData = sortedData.filter((item) => {
+      return Number(item.original_price) <= Number(state.range);
+    });
+  }
+  sortedData = sortedData.filter((item) => {
+    return item.rating >= state.ratings;
+  });
+  const isChecked = (value) => {
+    return value === true;
+  };
+  const condition = Object.values(state.selectedCategories).some(isChecked);
```

Comment:
You can directly do that `.some(value=> value)`

Reply:
done!!